File: python/bit-manipulation/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = 1
        if x < 0:
            sign = -1

        rev, x = 0, abs(x)
        while x > 0:
            x, mod = x // 10, x % 10
            rev *= 10
            rev += mod
            if rev > 2 ** 31 - 1:
                return 0

        return sign * rev

File: c++/bit-manipulation/reverse-integer.cc
class Solution {
public:
    int reverse(int x) {
        int sign = x < 0 ? -1 : 1;
        int answer = 0;
        if (x < -2147483647) {
            return 0;
        }
        int digits = std::abs(x);
        int limit = (std::pow(2, 31) - 1) / 10;
        while (digits > 0) {
            if (answer > limit) {
                return 0;
            }
            answer *= 10;
            answer += digits % 10;
            digits /= 10;
        }
        return sign * answer;
    }
};
